source/Octopus.Tentacle.Client/Scripts/ObservingScriptOrchestrator.cs

```diff
@@ -73,8 +73,7 @@ public async Task<ScriptExecutionResult> ExecuteScript(StartScriptCommandV3Alpha
             {
                 if (scriptExecutionCancellationToken.IsCancellationRequested)
                 {
-                    // We don't want to cancel this operation as it is responsible for stopping the script executing on the Tentacle
-                    lastStatusResponse = await Cancel(lastStatusResponse, CancellationToken.None).ConfigureAwait(false);
+                    lastStatusResponse = await Cancel(lastStatusResponse, scriptExecutionCancellationToken).ConfigureAwait(false);
```

Comment:
The fact CancellationToken.None was passed into the method from this abstract class made it hard to understand the flow in the implementations. The actual cancellation token is now passed, and the implementations have been changed to use CancellationToken.None instead of the passed cancellation token. where appropriate. This makes reading the logic in the implementation much clearer.